DragonLoop/DragonLoopApp/DragonLoopApp/Views/MapPage.xaml.cs

```diff
@@ -29,5 +32,41 @@ protected override void OnAppearing()
             if (viewModel.Routes == null || !viewModel.Routes.Any())
                 viewModel.LoadRoutesCommand.Execute(null);
         }
+
+        private async void Handle_Route_Toggle(object sender, Xamarin.Forms.ToggledEventArgs e)
+        {
+            //set up stop on screen
+            if (e.Value == true)
+            {
+                ViewCell cell = (sender as Switch).Parent.Parent as ViewCell;
+                Route route = cell.BindingContext as Route;
+
+                if (route != null)
+                {
+                    await viewModel.LoadRouteStops(route.RouteId);
+                    
+                    //Load pins onto map
+                    List<Stop> stops = new List<Stop>(viewModel.RouteStops);
+                    for(int i = 0; i < viewModel.RouteStops.Count(); i++)
```

Comment:
This can be:
`foreach (var stop in RouteStops)`
and then you won't need to define the stops list above. You should just be able to use stop directly instead of stops[i].